frontend/src/metabase/models/components/ModelDetailPage/ModelActionDetails/ModelActionDetails.tsx

```diff
@@ -58,26 +62,52 @@ function ModelActionDetails({
   actions,
   onEnableImplicitActions,
   onArchiveAction,
+  onDeleteAction,
 }: Props) {
+  const { show: askConfirmation, modalContent: confirmationModal } =
+    useConfirmation();
+
   const database = model.database();
   const hasActionsEnabled = database != null && database.hasActionsEnabled();
   const canWrite = model.canWriteActions();
-  const hasImplicitActions = actions.some(action => action.type === "implicit");
 
   const actionsSorted = useMemo(
     () => _.sortBy(actions, mostRecentFirst),
     [actions],
   );
 
+  const implicitActions = useMemo(
+    () => actions.filter(action => action.type === "implicit"),
+    [actions],
+  );
+
+  const onDeleteImplicitActions = useCallback(() => {
+    askConfirmation({
+      title: t`Disable basic actions`,
+      message: t`Disabling basic actions will also remove any buttons that use these actions. Are you sure you want to continue?`,
+      confirmButtonText: t`Continue`,
+      onConfirm: () => {
+        implicitActions.forEach(action => {
+          onDeleteAction(action);
+        });
+      },
+    });
+  }, [implicitActions, askConfirmation, onDeleteAction]);
+
   const menuItems = useMemo(() => {
+    const hasImplicitActions = implicitActions.length > 0;
     return [
       {
-        title: t`Create basic actions`,
+        title: hasImplicitActions
```

Comment:
nit: I think structuring the code like this makes it a little hard to understand at glance as everything is crunched into a single return value. I propose we return different values instead.

```ts
    if (hasImplicitActions) {
      return [
        {
          title: t`Disable basic actions`,
          icon: "bolt",
          action: onDeleteImplicitActions,
        },
      ];
    } else {
      return [
        {
          title: t`Create basic actions`,
          icon: "bolt",
          action: onEnableImplicitActions,
        },
      ];
    }
```

What do you think? This is a little longer, but hopefully clearer.